MetadataExtractor/IO/BufferReader.cs

```diff
@@ -135,4 +138,250 @@ public string GetString(int bytesRequested, Encoding encoding)
 
         return encoding.GetString(bytes);
     }
+
+    public StringValue GetNullTerminatedStringValue(int maxLengthBytes, Encoding? encoding = null, bool moveToMaxLength = false)
+    {
+        var bytes = GetNullTerminatedBytes(maxLengthBytes, moveToMaxLength);
+
+        return new StringValue(bytes, encoding);
+    }
+
+    public byte[] GetNullTerminatedBytes(int maxLengthBytes, bool moveToMaxLength = false)
```

Comment:
We may be able to return a ReadOnlySpan<byte> here, instead

Reply:
Possibly, with the consequence that StringValue will have to change.

Reply:
We'd still need to use ToArray when an array is actually needed, but there's a few cases where we would be able to utilize a span directly.

For instance, here:

```
var bytes = GetNullTerminatedBytes(index, maxLengthBytes);

return (encoding ?? Encoding.UTF8).GetString(bytes);
```

I did some exploration on whether to change StringValue to accept ReadOnlyMemory<byte>, which could eliminate some allocations at the risk of rooting larger byte[] segments. I'm not sure if it's worth the trade-off.